devices/Mcp7940xx/Mcp7940m.cs

```diff
@@ -0,0 +1,599 @@
+        /// </remarks>
+        public void SetGeneralPurposeOutputHigh()
+        {
+            RegisterHelper.SetRegisterBit(_I2cDevice, (byte)Register.Control, (byte)ControlRegister.GeneralPurposeOutput);
+        }
+
+        /// <summary>
+        /// Sets the MFP pin level to logic low.
+        /// </summary>
+        /// <remarks>
+        /// General purpose output can only be used when Alarm1, Alarm2 and square wave output are all disabled.
+        /// </remarks>
+        public void SetGeneralPurposeOutputLow()
+        {
+            RegisterHelper.ClearRegisterBit(_I2cDevice, (byte)Register.Control, (byte)ControlRegister.GeneralPurposeOutput);
+        }
+
+        /// <summary>
+        /// Checks if general purpose output is enabled.
+        /// </summary>
+        /// <returns>Returns <c>true</c> if general purpose output is enabled, <c>false</c> if not.</returns>
+        public bool GeneralPurposeOutputIsEnabled()
+        {
+            if (Alarm1IsEnabled() || Alarm2IsEnabled() || SquareWaveOutputIsEnabled())
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Checks if general purpose output is setting the state of the MFP pin to high.
+        /// </summary>
+        /// <returns>Returns <c>true</c> if MFP pin is high, <c>false</c> if not.</returns>
+        public bool GeneralPurposeOutputIsHigh()
+        {
+            return RegisterHelper.RegisterBitIsSet(_I2cDevice, (byte)Register.Control, (byte)ControlRegister.GeneralPurposeOutput);
+        }
+
+        #endregion
+
+        #region Square Wave
+
+        /// <summary>
+        /// Enables square wave output from the MFP pin.
+        /// </summary>
+        public void EnableSquareWaveOutput()
+        {
+            RegisterHelper.SetRegisterBit(_I2cDevice, (byte)Register.Control, (byte)ControlRegister.SquareWaveOutput);
+        }
+
+        /// <summary>
+        /// Disables square wave output from the MFP pin.
+        /// </summary>
+        public void DisableSquareWaveOutput()
+        {
+            RegisterHelper.ClearRegisterBit(_I2cDevice, (byte)Register.Control, (byte)ControlRegister.SquareWaveOutput);
+        }
+
+        /// <summary>
+        /// Checks if square wave output is enabled.
+        /// </summary>
+        /// <returns>Returns <c>true</c> if square wave is enabled, <c>false</c> if not.</returns>
+        public bool SquareWaveOutputIsEnabled()
+        {
+            return RegisterHelper.RegisterBitIsSet(_I2cDevice, (byte)Register.Control, (byte)ControlRegister.SquareWaveOutput);
+        }
+
+        /// <summary>
+        /// Gets the frequency of the square wave output on the MFP pin.
+        /// </summary>
+        /// <returns>A <see cref = "SquareWaveFrequency" /> object whose value is the current frequency of the square wave.</returns>
+        public SquareWaveFrequency GetSquareWaveFrequency()
```

Comment:
It can be. I usually only use properties when there is little or no logic and GetSquareWaveFrequencey requires a round trip to the device. If convention on nanoFramework is to use a property in a situation like this, most other Get/Set functions could be made properties as well to match. Let me know which way you would prefer and I can switch them over.